app/frontend/controllers/tab_controller.js

```diff
@@ -422,6 +437,38 @@ class TabController {
     uiHelper.configureButtonStyles('.ui-tabs-nav');
   }
 
+  saveTabScrollPosition(tabIndex) {
+    var metaTab = this.getTab(tabIndex);
+    var firstVerseListAnchor = uiHelper.getFirstVisibleVerseAnchor();
+
+    if (metaTab != null && firstVerseListAnchor != null) {
+      metaTab.setLocation(firstVerseListAnchor);
+    }
+  }
+
+  savePreviousTabScrollPosition() {
+    if (this.lastSelectedTabIndex != null) {
+      this.saveTabScrollPosition(this.lastSelectedTabIndex);
+    }
+  }
+
+  restoreScrollPosition(tabIndex) {
+    var metaTab = this.getTab(tabIndex);
+
+    if (metaTab != null) {
+      var currentVerseListFrame = app_controller.getCurrentVerseListFrame(tabIndex);
```

Comment:
Maybe `const currentVerseListFrame` would be better

Reply:
`const` on Objects or Arrays actually does not prevent modifying them. It merely fixes the variable to the object it is pointing to. With `const` I actually see the point more for numeric values or strings, not so much for Objects and Arrays.